sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -987,6 +985,24 @@ object RemoveRedundantSorts extends Rule[LogicalPlan] {
     case f: Filter => f.condition.deterministic
     case _ => false
   }
+
+  def isOrderIrrelevantAggs(aggs: Seq[NamedExpression]): Boolean = {
+    def isOrderIrrelevantAggFunction(func: AggregateFunction): Boolean = func match {
+      case _: Sum => true
+      case _: Min => true
+      case _: Max => true
+      case _: Count => true
+      case _: Average => true
+      case _: CentralMomentAgg => true
+      case _ => false
+    }
+
+    aggs.flatMap { e =>
```

Comment:
If `aggs` only has a single `PythonFunction`, it seems this method returns `true`?

Reply:
yes, I'll try to fix it

Reply:
Thanks for pointing out this. I've updated and add a UT for udf